tests/__init__.py

```diff
@@ -43,6 +44,32 @@ def run_long_tests() -> bool:
     return os.environ.get("XDP_TEST_RUN_LONG") is not None
 
 
+def check_program_success(cmd) -> bool:
+    proc = subprocess.Popen(
+        cmd, stdout=None, stderr=None, shell=True, universal_newlines=True
+    )
+    _ = proc.communicate()
+    return proc.returncode == 0
+
+
+def can_run_fuse() -> Tuple[bool, str]:
+    if not check_program_success("fusermount3 --version"):
+        return (False, "no fusermount3")
+
+    if not check_program_success(
+        "capsh --print | grep -q 'Bounding set.*[^a-z]cap_sys_admin'"
+    ):
+        return (False, "No cap_sys_admin in bounding set, can't use FUSE")
+
+    if not check_program_success("[ -w /dev/fuse ]"):
+        return (False, "no write access to /dev/fuse")
+
+    if not check_program_success("[ -e /etc/mtab ]"):
+        return (False, "no /etc/mtab")
+
+    return (True, "success")
```

Comment:
bit awkward with the generic tuple returned, you could write this as

```py
def fuse_check_failed() -> str | None:
   if ...:
      return "no fusermount3"

   return None
```
and in the caller:
```py
if error := fuse_check_failed():
   pytest.skip(f"{error}")
```
(walrus is python 3.8)

or even more pythonic would probably be a `ensure_fuse()` that throws a `FuseNotSupportedException`.